src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -31,6 +32,35 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
     void Profiles::OnNavigatedTo(const NavigationEventArgs& e)
     {
         _State = e.Parameter().as<Editor::ProfilePageNavigationState>();
+
+        const auto& colorSchemeMap{ _State.Schemes() };
+        for (const auto& pair : colorSchemeMap)
+        {
+            _ColorSchemeList.Append(pair.Key());
+        }
+    }
+
```

Comment:
if you navigate back to the same page, isn't this gonna make the color scheme list grow without bound?

Reply:
Surprisingly no. And I can't figure out why.